lib/improver/metadata/utilities.py

```diff
@@ -30,10 +30,64 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """General IMPROVER metadata utilities"""
 
+import dask
 import hashlib
+import iris
 # Usage of pickle in this module is only for creation of pickles and hashing
 # the contents. There is no loading pickles which would create security risks.
 import pickle  # nosec
+import numpy as np
+
+
+def create_new_diagnostic_cube(
+        name, units, coordinate_template, attributes=None, data=None):
+    """
+    Creates a template for a new diagnostic cube with suitable metadata.
+
+    Args:
+        name (str):
+            Standard or long name for output cube
+        units (str or cf_units.Unit):
+            Units for output cube
+        coordinate_template (iris.cube.Cube):
+            Cube from which to copy dimensional and auxiliary coordinates
+        attributes (dict or None):
+            Dictionary of attribute names and values
+        data (np.ndarray or None):
+            Data array.  If not set, cube is filled with zeros using a lazy
+            data object, as this will be overwritten later by the caller
+            routine.
+
+    Returns:
+        iris.cube.Cube:
+            Cube with correct metadata to accommodate new diagnostic field
+    """
+    if data is None:
+        data = dask.array.zeros_like(
+            coordinate_template, dtype=np.float32, chunks=-1)
+
+    aux_coords_and_dims, dim_coords_and_dims = [
```

Comment:
Nested list comprehensions can be pretty nasty (although this isn't bad). I think this functionality might actually be useful elsewhere, so I'd probably separate this out into a function:
```
def _coords_and_dims(cube, coord_type):
    coords = getattr(cube, coord_type)
    return [(coord, cube.coord_dims(coord))
            for coord in getattr(cube, coord_type)]
```

Reply:
Hmm I'm not sure.  Short functions are good, but I am 100% sure this is not being used elsewhere in IMPROVER, nor is it obviously going to be needed in the design we're going for.  The reason for this is that we really don't use `iris.cube.Cube` to initialise - we copy cubes everywhere - and where we shouldn't, this function is intended to replace those `copy` statements with a more correct initialisation.  So actually I prefer the original version.

Reply:
I disagree on the first point, but I'm not entirely wedded to the idea of separating it out (to be honest, I think this method should exist in Iris).
    A quick `grep` for `cube.coord_dims` points at at least one plcae which could potentially benefit from this function. e.g: `utilities/spatial.py` has a block of code that looks like this:
```
252         for coord in cube.dim_coords:                                           
253             dims = cube.coord_dims(coord)                                       
254             if coord.name() in [coord_name]:                                    
255                 coord = coord.copy(points=mean_points)                          
256             diff_cube.add_dim_coord(coord.copy(), dims)                         
257         for coord in cube.aux_coords:                                           
258             dims = cube.coord_dims(coord)                                       
259             diff_cube.add_aux_coord(coord.copy(), dims)                         
260         for coord in cube.derived_coords:                                       
261             dims = cube.coord_dims(coord)                                       
262             diff_cube.add_aux_coord(coord.copy(), dims) 
```
which could potentially be refactored to use this function in the future. (But I wasn't suggesting that we make it a public function, but instead a private function nested within the current one, just to try to separate it out). Like I said, I don't mind either way though.

On the second point, I agree that we should use this rather than copy cubes. :)

Reply:
(Agreed with @cgsandford to go with what's currently here and not create another function right now.)